classes/Product.php

```diff
@@ -4093,18 +4093,7 @@ public function getIdProductAttributeMostExpensive()
      */
     public function getDefaultIdProductAttribute()
     {
-        if (!Combination::isFeatureActive()) {
-            return 0;
-        }
-
-        return (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
-            '
-            SELECT pa.`id_product_attribute`
-            FROM `' . _DB_PREFIX_ . 'product_attribute` pa
-            ' . Shop::addSqlAssociation('product_attribute', 'pa') . '
-            WHERE pa.`id_product` = ' . (int) $this->id . '
-            AND product_attribute_shop.default_on = 1'
-        );
+        return (int) Product::getDefaultAttribute($this->id);
```

Comment:
I'm sorry bu to me the two methods don't have the same behaviour:
- `getDefaultIdProductAttribute` strictly returns the combination identified explicitly as the default one based on its `default_on`
- `getDefaultAttribute` has many fallback results in case the later is not found, it can use the first found combination, it also includes a check by quantity In your case the quantity used is 0 so most of the query specific stock related are not used But it doesn't change the fact that it tries several fallbacks for the same result

This method can be used to detect if the DB is inconsistent (meaning no default combination is actually defined) whereas the other can not due to its fallback strategies. It doesn't mean fallbacks are bad or good it's just two different needs.

Reply:
Good point @jolelievre